cpp/src/arrow/compute/kernels/scalar_arithmetic.cc

```diff
@@ -1491,6 +1500,8 @@ ArrayKernelExec ArithmeticExecFromOp(detail::GetTypeId get_id) {
     case Type::INT64:
     case Type::TIMESTAMP:
       return KernelGenerator<Int64Type, Int64Type, Op>::Exec;
+    case Type::DATE64:
+      return KernelGenerator<Int64Type, Date64Type, Op>::Exec;
```

Comment:
Instead of adding this case here (which I think would cause the compiler to instantiate this template for all the kernel structs, even ones we don't care about), how about just writing `auto exec_date_64 = ScalarBinaryEqualTypes<...>::Exec` below?

Reply:
Done.